### ISSUE TITLE
[Bug-Candidate]: Slither breaks on Foundry test utils usage

### ISSUE THREAD

**Issue by CodeSandwich** [crytic/slither#1422]
### Describe the issue:

When Slither analyzes a test using Foundry test utils, it crashes.

A hacky workaround is to hide the tests from Slither, e.g. with
```bash
FOUNDRY_TEST=nonexisten slither .
```

### Code example to reproduce the issue:

https://github.com/radicle-dev/drips-contracts

### Version:

0.8.3

### Relevant log output:

```shell
'forge build --extra-output abi --extra-output userdoc --extra-output devdoc --extra-output evm.methodIdentifiers --force' running
Compiling 52 files with 0.8.17
Solc 0.8.17 finished in 7.19s
Compiler run successful

Traceback (most recent call last):
  File "/home/zuczek/workspace/slither/slither/core/cfg/node.py", line 911, in _find_read_write_call
    self._high_level_calls.append((ir.destination.type.type, ir.function))
AttributeError: 'NoneType' object has no attribute 'type'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/home/zuczek/workspace/slither/slither/__main__.py", line 826, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/home/zuczek/workspace/slither/slither/__main__.py", line 97, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/home/zuczek/workspace/slither/slither/__main__.py", line 75, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/home/zuczek/workspace/slither/slither/slither.py", line 130, in __init__
    parser.analyze_contracts()
  File "/home/zuczek/workspace/slither/slither/solc_parsing/slither_compilation_unit_solc.py", line 519, in analyze_contracts
    self._convert_to_slithir()
  File "/home/zuczek/workspace/slither/slither/solc_parsing/slither_compilation_unit_solc.py", line 713, in _convert_to_slithir
    func.generate_slithir_and_analyze()
  File "/home/zuczek/workspace/slither/slither/core/declarations/function.py", line 1705, in generate_slithir_and_analyze
    node.slithir_generation()
  File "/home/zuczek/workspace/slither/slither/core/cfg/node.py", line 721, in slithir_generation
    self._find_read_write_call()
  File "/home/zuczek/workspace/slither/slither/core/cfg/node.py", line 914, in _find_read_write_call
    raise SlitherException(
slither.exceptions.SlitherException: Function not found on IR: TMP_320(None) = HIGH_LEVEL_CALL, dest:TMP_319(None), function:sig, arguments:['1889567281']  .
Node: EXPRESSION stdstore.target(token).sig(0x70a08231).with_key(to).checked_write(give) (lib/forge-std/src/Test.sol#104-108)
Function: tip
Please try compiling with a recent Solidity version. 'NoneType' object has no attribute 'type'
Error:
Function not found on IR: TMP_320(None) = HIGH_LEVEL_CALL, dest:TMP_319(None), function:sig, arguments:['1889567281']  .
Node: EXPRESSION stdstore.target(token).sig(0x70a08231).with_key(to).checked_write(give) (lib/forge-std/src/Test.sol#104-108)
Function: tip
Please try compiling with a recent Solidity version. 'NoneType' object has no attribute 'type'
Please report an issue to https://github.com/crytic/slither/issues
```

**Comment by elopez** [comment]
Thanks for the report! I see you're using an older version of Slither; I'm unable to reproduce that crash on Slither 0.9.0, but there is another error reported which is being tracked in https://github.com/crytic/slither/issues/1352

```
$ slither .
'forge build --extra-output abi --extra-output userdoc --extra-output devdoc --extra-output evm.methodIdentifiers --force' running
Missing dependencies found. Installing now...

Compiling 52 files with 0.8.17
Solc 0.8.17 finished in 49.06s
Compiler run successful

Traceback (most recent call last):
  File "/usr/lib/python3.10/site-packages/slither/__main__.py", line 826, in main_impl
    ) = process_all(filename, args, detector_classes, printer_classes)
  File "/usr/lib/python3.10/site-packages/slither/__main__.py", line 97, in process_all
    ) = process_single(compilation, args, detector_classes, printer_classes)
  File "/usr/lib/python3.10/site-packages/slither/__main__.py", line 75, in process_single
    slither = Slither(target, ast_format=ast, **vars(args))
  File "/usr/lib/python3.10/site-packages/slither/slither.py", line 102, in __init__
    parser.parse_top_level_from_loaded_json(ast, path)
  File "/usr/lib/python3.10/site-packages/slither/solc_parsing/slither_compilation_unit_solc.py", line 320, in parse_top_level_from_loaded_json
    raise SlitherException(f"Top level {top_level_data[self.get_key()]} not supported")
slither.exceptions.SlitherException: Top level UsingForDirective not supported
Error:
Top level UsingForDirective not supported
Please report an issue to https://github.com/crytic/slither/issues

```

Removing the unsupported directive as follows makes slither work:

```diff
diff --git a/src/Drips.sol b/src/Drips.sol
index 9e60019..abb864f 100644
--- a/src/Drips.sol
+++ b/src/Drips.sol
@@ -44,7 +44,7 @@ struct DripsHistory {
 /// If zero, drip until balance runs out.
 type DripsConfig is uint256;
 
-using DripsConfigImpl for DripsConfig global;
+//using DripsConfigImpl for DripsConfig global;
 
 library DripsConfigImpl {
     /// @notice Create a new DripsConfig.
@@ -101,6 +101,7 @@ library DripsConfigImpl {
 /// It's up to the caller to guarantee that this limit is never exceeded,
 /// failing to do so may result in a total protocol collapse.
 abstract contract Drips {
+    using DripsConfigImpl for DripsConfig;
     /// @notice Maximum number of drips receivers of a single user.
     /// Limits cost of changes in drips configuration.
     uint256 internal constant _MAX_DRIPS_RECEIVERS = 100;
diff --git a/src/test/Drips.t.sol b/src/test/Drips.t.sol
index f0f5d8a..0b62960 100644
--- a/src/test/Drips.t.sol
+++ b/src/test/Drips.t.sol
@@ -23,6 +23,8 @@ contract PseudoRandomUtils {
 }
 
 contract DripsTest is Test, PseudoRandomUtils, Drips {
+    using DripsConfigImpl for DripsConfig;
+
     bytes internal constant ERROR_NOT_SORTED = "Receivers not sorted";
     bytes internal constant ERROR_INVALID_DRIPS_LIST = "Invalid current drips list";
     bytes internal constant ERROR_TIMESTAMP_EARLY = "Timestamp before last drips update";

```

**Comment by CodeSandwich** [comment]
Oh, that's really cool! Because of this directive I was using branch `dev-fix-usingfor` from PR https://github.com/crytic/slither/pull/1378 which doesn't seem to based off 0.9.0, so I've apparently pushed myself into a feature limbo. If you're saying that it's fixed in 0.9.0, I'll close this issue and wait for `dev-fix-usingfor` to be updated or merged. Thank you for looking into this!
